spark-jobs/src/main/scala/za/co/absa/enceladus/common/CommonJobExecution.scala

```diff
@@ -52,7 +51,7 @@ trait CommonJobExecution {
   protected case class PreparationResult(dataset: Dataset,
                                          reportVersion: Int,
                                          pathCfg: PathConfig,
-                                         s3Config: S3Config,
+                                         fileSystems: FileSystems,
```

Comment:
Thought: I feel like `pathCfg` and `fileSystems` could be one object.

Reply:
Yes, I found a way, I believe it is a tad better now.